orderby-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/CompositeKey.java

```diff
@@ -0,0 +1,198 @@
+                                           "String, Int, Long, Float, Double, Byte");
+    }
+    return res;
+  }
+
+  @Override
+  public void write(DataOutput dataOutput) throws IOException {
+    dataOutput.writeUTF(structureRecordJSON);
+    dataOutput.writeUTF(sortFieldsJSON);
+    dataOutput.writeUTF(schemaJSON);
+  }
+
+  @Override
+  public void readFields(DataInput dataInput) throws IOException {
+    this.structureRecordJSON = dataInput.readUTF();
+    this.sortFieldsJSON = dataInput.readUTF();
+    this.schemaJSON = dataInput.readUTF();
+  }
+
+  /**
+   * This comparator controls the sort order of the keys.
+   *
+   * @param compositeKeyToCompare Compositekey for comparison
+   * @return structured records sorted on the keys
+   */
+  @Override
+  public int compareTo(CompositeKey compositeKeyToCompare) {
+    int res = 0;
+    Schema schema = gson.fromJson(schemaJSON, schemaType);
+    try {
+      StructuredRecord structuredRecord1 = StructuredRecordStringConverter
+        .fromJsonString(this.structureRecordJSON, schema);
+      StructuredRecord structuredRecord2 = StructuredRecordStringConverter
+        .fromJsonString(compositeKeyToCompare.structureRecordJSON, schema);
+      LinkedHashMap<String, String> sortFields = gson.fromJson(sortFieldsJSON, mapType);
+
+      for (Map.Entry<String, String> entry : sortFields.entrySet()) {
+        String key = entry.getKey();
+        String value = entry.getValue();
+        if (key.contains(FIELD_DELIMITER)) {
+          res = compareNestedRecords(key, value, structuredRecord1,
+                                     structuredRecord2);
+        } else {
+          res = compareRecords(key, value, structuredRecord1, structuredRecord2);
+        }
+        if (res != 0) {
+          return res;
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Error reading structured record : " + e.getMessage(), e);
+    }
+    return res;
+  }
+
+  /**
+   * Get sorting order for nested structured records using recursive call
+   *
+   * @param key               contains the value to sort the data
+   * @param value             determines whether the objects are to be sorted in ascending or descending order for key
+   * @param structuredRecord1 first Structured record
+   * @param structuredRecord2 second Structured record to compare against
+   * @return structured records sorted on the keys
+   */
+  private int compareNestedRecords(String key, String value, StructuredRecord structuredRecord1,
+                                   StructuredRecord structuredRecord2) {
+
+    if (key.contains(FIELD_DELIMITER)) {
+      String[] nestedRecord = key.split(FIELD_DELIMITER);
+      Schema.Field field = structuredRecord1.getSchema().getField(nestedRecord[0].trim());
+      Schema.Type type = field.getSchema().getType();
+      if (type.equals(Schema.Type.RECORD)) {
```

Comment:
The if(#172) statement checks, if the path specified by user for selecting the nested field is valid, by checking the parent is of type RECORD.
For the current implementation we are only supporting simple types with nullable (as UNION): String, Int, Long, Short, Float, Double, Byte as mentioned in the use-case.